coalib/bearlib/abstractions/ExternalBearWrap.py

```diff
@@ -0,0 +1,174 @@
+import json
+from functools import partial
+from collections import OrderedDict
+
+from coalib.bears.LocalBear import LocalBear
+from coala_decorators.decorators import enforce_signature
+from coalib.misc.Shell import run_shell_command
+from coalib.results.Diff import Diff
+from coalib.results.Result import Result
+from coalib.results.RESULT_SEVERITY import RESULT_SEVERITY
+from coalib.settings.FunctionMetadata import FunctionMetadata
+
+
+def _prepare_options(options):
+    """
+    Prepares options for a given options dict in-place.
+
+    :param options: The options dict that contains user/developer inputs.
+    :raises ValueError: Raised when illegal options are specified.
+    """
+    allowed_options = {"executable",
+                       "settings"}
+
+    # Check for illegal superfluous options.
+    superfluous_options = options.keys() - allowed_options
+    if superfluous_options:
+        raise ValueError(
+            "Invalid keyword arguments provided: " +
+            ", ".join(repr(s) for s in sorted(superfluous_options)))
+
+
+def _create_wrapper(klass, options):
+    class ExternalBearWrapBase(LocalBear):
+
+        @staticmethod
+        def create_arguments():
+            """
+            This method has to be implemented by the class that uses
+            the decorator in order to create the arguments needed for
+            the executable.
+            """
+            raise NotImplementedError
+
+        @classmethod
+        def get_executable(cls):
+            """
+            Returns the executable of this class.
+
+            :return: The executable name.
+            """
+            return options["executable"]
+
+        @staticmethod
+        def normalize_desc(val):
+            """
+            Normalizes the description of the parameters only if there
+            is none provided.
+
+            :return: A value for the OrderedDict in the FunctionMetada object.
+            """
+            if val[0] != "":
+                return val
+            desc = "No description given."
+            if len(val) == 2:
+                return (desc, val[1])
+            else:
+                desc += " Defaults to " + str(val[2])
+                return (desc, val[1], val[2])
```

Comment:
The settings here are passed in the decorator (I will make a test for that too), so I have to manually craft the FunctionMetadata object.